pkg/liqoctl/purge/handler.go

```diff
@@ -0,0 +1,468 @@
+		return err
+	}
+
+	var nodeList corev1.NodeList
+	if err := h.cl.List(ctx, &nodeList, client.MatchingLabels{
+		consts.RemoteClusterID: h.remoteClusterIdentity.ClusterID}); err != nil {
+		s.Fail(fmt.Sprintf("Failed to list nodes: %v", err))
+		return err
+	}
+
+	for i := range nodeList.Items {
+		node := &nodeList.Items[i]
+		node.Finalizers = []string{}
+		if err := client.IgnoreNotFound(h.cl.Update(ctx, node)); err != nil {
+			s.WarningPrinter.Println(fmt.Sprintf("Failed to remove finalizers from node %s: %v", node.Name, err))
+			continue
+		}
+	}
+
+	s.Success("Deleted virtual node")
+	return nil
+}
+
+func pointerList[T any](src []T) []*T {
+	dst := make([]*T, len(src))
+	for i := range src {
+		dst[i] = &src[i]
+	}
+	return dst
+}
+
+func forceDeleteNetworkConfigs(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
+	return forceDelete(ctx, cl, namespace, s, &netv1alpha1.NetworkConfig{}, &netv1alpha1.NetworkConfigList{},
+		func(list *netv1alpha1.NetworkConfigList) []*netv1alpha1.NetworkConfig { return pointerList(list.Items) })
+}
+
+func forceDeleteResourceRequests(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
+	return forceDelete(ctx, cl, namespace, s, &discoveryv1alpha1.ResourceRequest{}, &discoveryv1alpha1.ResourceRequestList{},
+		func(list *discoveryv1alpha1.ResourceRequestList) []*discoveryv1alpha1.ResourceRequest {
+			return pointerList(list.Items)
+		})
+}
+
+func forceDeleteResourceOffers(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
+	return forceDelete(ctx, cl, namespace, s, &sharingv1alpha1.ResourceOffer{}, &sharingv1alpha1.ResourceOfferList{},
+		func(list *sharingv1alpha1.ResourceOfferList) []*sharingv1alpha1.ResourceOffer {
+			return pointerList(list.Items)
+		})
+}
+
+func forceDeleteNamespaceMaps(ctx context.Context, cl client.Client, namespace string, s *pterm.SpinnerPrinter) error {
+	return forceDelete(ctx, cl, namespace, s, &virtualkubeletv1alpha1.NamespaceMap{}, &virtualkubeletv1alpha1.NamespaceMapList{},
+		func(list *virtualkubeletv1alpha1.NamespaceMapList) []*virtualkubeletv1alpha1.NamespaceMap {
+			return pointerList(list.Items)
+		})
+}
+
+func forceDelete[T client.Object, U client.ObjectList](ctx context.Context, cl client.Client,
+	namespace string, s *pterm.SpinnerPrinter,
+	obj T, objList U, objListToObjArray func(U) []T) error {
+
+	if err := cl.DeleteAllOf(ctx, obj, client.InNamespace(namespace)); err != nil {
+		return err
+	}
+
+	if err := cl.List(ctx, objList, client.InNamespace(namespace)); err != nil {
+		return err
+	}
+
+	for _, obj := range objListToObjArray(objList) {
+		obj.SetFinalizers([]string{}) // nolint:typecheck // golangci is not able to understand it at the moment https://github.com/golangci/golangci-lint/issues/2649
+		if err := client.IgnoreNotFound(cl.Update(ctx, obj)); err != nil {
+			s.WarningPrinter.Printf("%v\n", err)
```

Comment:
Should probably enrich the error a bit, to make it easier to understand. Should also probably return the error.